src/main/thrift/parquet.thrift

```diff
@@ -1165,6 +1317,62 @@ struct FileMetaData {
   9: optional binary footer_signing_key_metadata
 }
 
+/** Metadata for a column in this file. */
+struct FileColumnMetadataV3 {
+  /** All column chunks in this file (one per row group) **/
+  1: required list<ColumnChunkV3> columns
+
+  /** Sort order used for the Statistics min_value and max_value fields
+   **/
+  2: optional ColumnOrder column_order;
+
+  /** NEW from v1: Optional key/value metadata for this column at the file level
+   **/
+  3: optional list<KeyValue> key_value_metadata
+}
+
+struct FileMetaDataV3 {
+  /** Version of this file **/
+  1: required i32 version
+
+  /** Parquet schema for this file **/
+  2: required list<SchemaElementV3> schema;
```

Comment:
Lists cannot really be decoded in a random access manner.  I suggest in V3 we should consider moving any list elements to a Page that has type list<byte> where each element is a serialized struct (thrift or something else if we choose to move away from it).

Reply:
For heavily nested nested lists we might want to separate type specific fields.

Reply:
Does it matter here?

Reply:
No, not here.  I think I need to do a more formal auditing to see what makes sense.

Reply:
skip lists do improve list random IO perf, or some variant where as the element list is built up somehow an offset to a later list element is added. But I don't know of any easy way to do that here.

Reply:
My concrete suggestion is to introduce a page encoding that supports random access: https://github.com/apache/parquet-format/pull/250 which IIUC is similar to the approach described here for Columns but allows for the solution to be more easily generalized.